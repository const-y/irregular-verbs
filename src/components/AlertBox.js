import React from 'react';
import { Alert } from 'react-bootstrap';

function AlertBox({ sampler, success, error }) {
  if (success) {
    return <Alert variant="success">Верно!</Alert>;
  }
  if (error) {
    return <Alert variant="danger">{error}</Alert>;
  }

  return <Alert variant="primary">{sampler[3]}</Alert>;
}

export default AlertBox;
