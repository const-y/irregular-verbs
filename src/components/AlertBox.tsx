import React from 'react';
import { Alert } from 'react-bootstrap';
import { DictionaryItem } from '../models/DictionaryItem';

interface AlertBoxProps {
  sampler: DictionaryItem;
  success: boolean;
  error: string;
}

function AlertBox({ sampler, success, error }: AlertBoxProps) {
  if (success) {
    return <Alert variant="success">Верно!</Alert>;
  }
  if (error) {
    return <Alert variant="danger">{error}</Alert>;
  }

  return <Alert variant="primary">{sampler[3]}</Alert>;
}

export default AlertBox;
