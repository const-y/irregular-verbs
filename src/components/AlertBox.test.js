import React from 'react';
import renderer from 'react-test-renderer';
import AlertBox from './AlertBox';

it('renders correctly', () => {
  const tree = renderer
    .create(<AlertBox sampler={['meet', 'met', 'met', 'встречать']}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});