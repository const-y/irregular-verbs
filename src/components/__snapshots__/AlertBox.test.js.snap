// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fade alert alert-primary show"
  role="alert"
>
  встречать
</div>
`;
